grafe-cli/src/generate/generate.ts

```diff
@@ -3,83 +3,176 @@ import yargs, { Options } from 'yargs';
-                },
-                {
-                    type: 'list',
-                    name: 'method',
-                    message: 'Which HTTP-Method should the new route be',
-                    choices: ['GET', 'POST', 'PUT', 'DELETE'],
-                },
-                {
-                    type: 'checkbox',
-                    message: 'Select the middlewares for this route',
-                    name: 'middlewares',
-                    choices: data.middlewares
-                }
-            ];
-
-            if (questions[2].choices == undefined || questions[2]?.choices?.length == 0) {
-                questions.splice(2, 1);
-            }
+    if (answers.type.toLowerCase() === 'route') {
+
+        const rootDir = await pkgDir(process.cwd());
+
+        let raw;
+        try {
+            raw = fs.readFileSync(path.join(rootDir, 'grafe.json'));
+        } catch (err) {
+            return console.log("The grafe command must be used within a grafe project.");
+        }
+
+        let data = JSON.parse(raw.toString());
+
+        let questions = [];
+        if (argv.routePath == undefined) {
+            questions.push({
+                type: 'input',
+                name: 'path',
+                message: 'How is the new route called'
+            });
+        }
 
-            inquirer.prompt(questions).then(async (answers) => {
-                generateRoute(answers.path, answers.method, answers.middlewares);
+        if (argv.method == undefined) {
+            questions.push({
+                type: 'list',
+                name: 'method',
+                message: 'Which HTTP-Method should the new route be',
+                choices: ['GET', 'POST', 'PUT', 'DELETE'],
             });
-        } else if (answers.type === "Middleware") {
-            inquirer.prompt([{
+        }
+
+        if (argv.middlewares == undefined) {
+            questions.push({
+                type: 'checkbox',
+                message: 'Select the middlewares for this route',
+                name: 'middlewares',
+                choices: data.middlewares
+            });
+
+            let index = questions.findIndex((item, i) => {
+                return item.name === 'middlewares'
+            });
+
+            if (questions[index].choices == undefined || questions[index]?.choices?.length == 0) {
+                questions.splice(index, 1);
+            }
+        }
+
+        if (questions.length > 0) {
+            answers = await inquirer.prompt(questions);
+        }
+
+        if (answers.path == undefined) {
+            answers.path = argv.routePath;
+        }
+
+        if (answers.method == undefined) {
+            answers.method = argv.method;
+        }
+
+        if (answers.middlewares == undefined) {
+            answers.middlewares = argv.middlewares;
+        }
+
+        generateRoute(answers.path, answers.method, answers.middlewares);
+    } else if (answers.type.toLowerCase() === "middleware") {
```

Comment:
Ich würd die zwei typen in zwei verschiedene Funktionen hauen. Das ma ned so lange funktionen haben